### PR TITLE
AOV denoise features and a few ninja fixes

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

**Comment by jdent02** [comment]
This should work better :)

**Comment by jdent02** [comment]
Snuck another commit in on this one.  Incremental mode is not automatically enabled in IPR mode so I added that in.

**Comment by jdent02** [comment]
Another ninja fix: Image readback during internal rendering only works when incremental mode is on.

File: properties.py [hunk 1]
```diff
@@ -264,6 +264,16 @@ def update_type(self, context):
         name="AOV name",
         description="Name of the built in AOV")
 
+    denoise_aov = BoolProperty(
+        name="Format AOV for denoising",
+        description="If checked this pass will be export as an individual file and properly formatted for use by the denoise utility",
+        default=False)
+
+    exclude_from_multi = BoolProperty(
```
Comment (bsavery) [inline properties.py]: I'm not sure why you'd want to do this?  But I guess it works.

File: engine.py [hunk 2]
```diff
@@ -408,7 +408,7 @@ def update_image():
             base, ext = render_output.rsplit('.', 1)
             # denoise data has the name .denoise.exr
             denoise_options = "-t%d" % self.rm.threads
-            denoise_data = base + '.denoise.' + 'exr'
+            denoise_data = base + '.variance.' + 'exr'
```
Comment (bsavery) [inline engine.py]: why the change from denoise to variance?

**Comment by bsavery** [comment]
So good things.  But why the change from .denoise in filename to .variance

Also it looks like the aov files get .denoiseable? Are they automatically run through denoise?

File: properties.py (hunk 1 above)
Comment (jdent02) [inline properties.py]: I put that description in before my latest changes so it's not quite accurate.  I'll change it

File: engine.py (hunk 2 above)
Comment (jdent02) [inline engine.py]: the denoise multilayer EXR was renamed to .variance. So this is to maintain internal denoising ability

**Comment by jdent02** [comment]
The denoise aov feature asks for a multilayer file named .variance that contains all the data passes needed.  The data passes are exactly the same as the normal beauty pass denoise.  You can use 'regular' denoise on the .variance named file without any issues, but the file must be named that to work with pass denoising.  That was my reasoning for the change.

**Comment by jdent02** [comment]
And no, the passes are not run through denoise automatically. I added that suffix to help a user see at a glance what individual passes are compatible with aov denoising while also allowing them to use the layer names they assigned in the aov panel.

**Comment by bsavery** [comment]
Ok.  Sounds good!  Thanks for clarifying


On Wed, May 11, 2016 at 4:05 PM, Jon D <notifications@github.com> wrote:

> And no, the passes are not run through denoise automatically. I added that
> suffix to help a user see at a glance what individual passes are compatible
> with aov denoising while also allowing them to use the layer names they
> assigned in the aov panel.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/pull/258#issuecomment-218616045>
>



-- 
brian.savery@gmail.com
508-274-8700

File: properties.py (hunk 1 above)
Comment (bsavery) [inline properties.py]: No I mean "Why would a user want to make an aov but not put it in the multilayer?  Not saying it hurst anything just not sure it's needed.

Comment (jdent02) [inline properties.py]: i didn't have a particular example in mind but I thought it would give a user more flexibility on how their passes and images are exported.

Comment (jdent02) [inline properties.py]: Or I just got carried away......  Not quite sure which

Comment (bsavery) [inline properties.py]: Yeah would be nice if the pass editor was more flexible!


On Wed, May 11, 2016 at 4:14 PM, Jon D <notifications@github.com> wrote:

> In properties.py
> <https://github.com/bsavery/PRMan-for-Blender/pull/258#discussion_r62942839>
> :
>
> > @@ -264,6 +264,16 @@ def update_type(self, context):
> >          name="AOV name",
> >          description="Name of the built in AOV")
> >
> > +    denoise_aov = BoolProperty(
> > +        name="Format AOV for denoising",
> > +        description="If checked this pass will be export as an individual file and properly formatted for use by the denoise utility",
> > +        default=False)
> > +
> > +    exclude_from_multi = BoolProperty(
>
> Or I just got carried away...... Not quite sure which
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/pull/258/files/57beb09857478d08591b5b665692f756d9eb80f3#r62942839>
>



-- 
brian.savery@gmail.com
508-274-8700

Comment (jdent02) [inline properties.py]: To that end, maybe as a long term project I'll try to implement a section in the output panel where a user can define one or more multilayer files and then assign passes to them......  Definitely above my skill set at the moment so it will take me some time to get it done.
